NotationV0/src/lib/runner.ts

```diff
@@ -24,3 +28,29 @@ export async function notationRunner(artifactRefs: string[], runCommand: (artifa
         throw new Error(taskLib.loc('FailedArtifacts', failedArtifactRefs.join(', ')));
     }
 }
+
+// WarningStream is a writable stream that extracts warnings from logs.
+class WarningStream extends Writable {
+    private buffer: string;
+
+    constructor() {
+        super();
+        this.buffer = '';
+    }
+
+    _write(chunk: Buffer, encoding: string, callback: () => void) {
+        this.buffer += chunk.toString();
+        const lines = this.buffer.split('\n');
+        this.buffer = lines.pop() || '';
+        // extract warnings related to security from logs
+        for (const line of lines) {
+            if (line.startsWith('Warning: Always sign the artifact using the digest')) {
+                taskLib.warning(line);
```

Comment:
quick question, in this scenario we are just doing `taskLib.warning(line)`, do we also need `console.log(line)`?

Reply:
The warning will also be shown in the logs, so I think we don't need to log it again.